ccm_web/server/src/canvas/canvas.scopes.ts

```diff
@@ -0,0 +1,13 @@
+/*
+Canvas API scopes, separated by privilege level
+
+Above each scope string is a comment indicating under which scope group or dropdown the scope
+can be found in the API Developer Key interface in Canvas.
+*/
+
+const privilegeLevelOneScopes = [
+  // Courses
+  'url:PUT|/api/v1/courses/:id'
```

Comment:
So we will end up putting all the API's we will be using for CCM here?

we might need to handle the case when we keep on adding API's in the scope then the exiting canvas token get invalidated and we need to generate a new token with repeat authorization flow. This use case needs to be verified before implementing this.  
**Nothing to do here. expressing my thoughts.**

Reply:
Yeah, good thought. I don't think this is going to happen very frequently, but we could delete canvas tokens preemptively if the scopes are going to change.

Reply:
> Why is it needed to have it in 2 place one in Canvas API key and another in the Application files?

What do you see as the alternative to this, @pushyamig? In the OAuth login documentation (https://canvas.instructure.com/doc/api/file.oauth_endpoints.html#get-login-oauth2-auth), it says if you use scopes you have to provide them as a parameter (otherwise you get `invalid_scope`). I'm not currently aware of another way to obtain those scopes in the backend without having them defined in a file. UDOIT does it in a similar way, looks like: https://github.com/ucfopen/UDOIT/blob/66a5e6d92015a459113d9696eba04aebf4226ae4/config/settings.php#L65-L92

> I noticed in the API key configuration something like this `Allow Include Parameters` do you know how to incorporate that in the canvas.scopes.ts? or if we need to?

I haven't thought too deeply about the include parameter option, but I think we should try to avoid using it if possible. It often provides data about related entities, and I'm thinking at this point that we'll be dealing with the core entities related to each scope. If we end up needing it, we can cross that bridge when we come to it.

Reply:
> What do you see as the alternative to this, @pushyamig?
I am not opposing to what is done if Canvas implemented it that way. I just wanted to know if that is needed. I did not wanted to read the documentation for canvas OAuth, it's easy to ask question here.
Seems silly though from Canvas design perspective we have configure it in Canvas API key and in the APP w.r.t scopes. On Bright side we know all the scoped from the code base

Reply:
I've created this issue to handle the issue of changing scopes: https://github.com/tl-its-umich-edu/canvas-course-manager-next/issues/104

Reply:
I think to allow this, we would just need to check that box; I don't think there is a change to the scope that's needed. We'd need to test that. However, I wonder if we can examine some of the calls that you have using `include` and change the endpoint (for instance from course to section), so the more specific scope can be used instead. This will need more thinking; I think we should figure this out incrementally as we address each new feature, not all at once.

Reply:
> I think we should figure this out incrementally as we address each new feature, not all at once.

I concur